.github/workflows/test_devdeps.yml

```diff
@@ -0,0 +1,73 @@
+name: test with development versions
+
+on:
+  push:
+    branches:
+      - master
+      - '*.x'
+    tags:
+      - '*'
+  pull_request:
```

Comment:
Should this include "synchronize" "labeled" or some combination of the 2 to allow an existing PR that has the label added to run the workflow (similar to the downstream label and job in asdf: https://github.com/asdf-format/asdf/blob/96800153294b61c271bbf6608ba509cc239dc314/.github/workflows/downstream.yml#L8-L10)?